### PR TITLE
Cleanup.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

File: libevmasm/LinkerObject.cpp [hunk 1]
```diff
@@ -38,7 +38,7 @@ void LinkerObject::link(map<string, h160> const& _libraryAddresses)
 	std::map<size_t, std::string> remainingRefs;
 	for (auto const& linkRef: linkReferences)
 		if (h160 const* address = matchLibrary(linkRef.second, _libraryAddresses))
-			address->ref().copyTo(ref(bytecode).cropped(linkRef.first, 20));
```
Comment (chriseth) [inline libevmasm/LinkerObject.cpp:41]: I noticed that this line is not tested. This has to be changed before this PR is merged.

File: libevmasm/LinkerObject.cpp [hunk 2]
```diff
@@ -38,7 +38,7 @@ void LinkerObject::link(map<string, h160> const& _libraryAddresses)
...
+			copy(address->data(), address->data() + 0x20, bytecode.begin() + linkRef.first);
```
Comment (chriseth) [inline libevmasm/LinkerObject.cpp:41]: And `0x20` has to be changed back to `20`.

File: libsolidity/ast/ASTPrinter.cpp [hunk 3]
```diff
@@ -579,8 +582,11 @@ void ASTPrinter::printSourcePart(ASTNode const& _node)
 	if (!m_source.empty())
 	{
 		SourceLocation const& location(_node.location());
-		*m_ostream << indentation() << "   Source: "
-				   << escaped(m_source.substr(location.start, location.end - location.start), false) << endl;
+		*m_ostream <<
```
Comment (axic) [inline libsolidity/ast/ASTPrinter.cpp:585]: I think for reviewability we should split this PR up. One simple change is replacing `escaped` with `jsonCompactPrint` and removing `escaped`.

File: libdevcore/FixedHash.h [hunk 4]
```diff
@@ -147,54 +121,17 @@ class FixedHash
...
 	inline unsigned firstBitSet() const
 	{
 		unsigned ret = 0;
 		for (auto d: m_data)
 			if (d)
+			{
 				for (;; ++ret, d <<= 1)
 					if (d & 0x80)
 						return ret;
-					else {}
```
Comment (axic) [inline libdevcore/FixedHash.h:197]: This could be nominated for most useful else statement ever.

File: libdevcore/FixedHash.h [hunk 5]
```diff
@@ -91,35 +85,15 @@ class FixedHash
 
 	// The obvious comparison operators.
 	bool operator==(FixedHash const& _c) const { return m_data == _c.m_data; }
-	bool operator!=(FixedHash const& _c) const { return m_data != _c.m_data; }
 	bool operator<(FixedHash const& _c) const { for (unsigned i = 0; i < N; ++i) if (m_data[i] < _c.m_data[i]) return true; else if (m_data[i] > _c.m_data[i]) return false; return false; }
```
Comment (axic) [inline libdevcore/FixedHash.h:88]: Where is this one used? If this comparison is needed I'd keep the other 3 too.

File: libdevcore/FixedHash.h [hunk 6]
```diff
@@ -91,35 +85,15 @@ class FixedHash
 
 	// The obvious comparison operators.
 	bool operator==(FixedHash const& _c) const { return m_data == _c.m_data; }
-	bool operator!=(FixedHash const& _c) const { return m_data != _c.m_data; }
```
Comment (axic) [inline libdevcore/FixedHash.h:94]: I think this comparison probably makes sense to keep.

File: libdevcore/FixedHash.h (hunk 5 above)
Comment (chriseth) [reply libdevcore/FixedHash.h:88]: If you use `FixedHash` in any sorted STL container, for example `std::map`, you need `operator<` (and only that).

Comment (axic) [reply libdevcore/FixedHash.h:88]: In that case I'd leave a comment explaining this reason.

File: libsolidity/ast/ASTPrinter.cpp (hunk 3 above)
Comment (axic) [reply libsolidity/ast/ASTPrinter.cpp:585]: Having a second look, are you sure the substring from the source is JSON?

File: libevmasm/LinkerObject.cpp (hunk 1 above)
Comment (chriseth) [reply libevmasm/LinkerObject.cpp:41]: Actually it should have been tested, there are end-to-end tests that perform linking. Running locally to see.

**Comment by axic** [comment]
Can we close this in favour of the split out versions?

File: libsolidity/ast/ASTPrinter.cpp (hunk 3 above)
Comment (chriseth) [reply libsolidity/ast/ASTPrinter.cpp:585]: No, it is not json, it is a pure string, but the json formatter will escape it in exactly the same way it was escaped before.

Comment (axic) [reply libsolidity/ast/ASTPrinter.cpp:585]: That's weird, because string on the root level is invalid according to the JSON spec.

Comment (chriseth) [reply libsolidity/ast/ASTPrinter.cpp:585]: This is just a restriction of the parser. Json::Value is an arbitrary json value, including any internal value.

**Pull request closed without merging**